### DISCUSSION TITLE
Deadlock when trying to deploy a small migration PG

### DISCUSSION THREAD

**Discussion by MorenoMdz** [prisma/prisma#21521] [Q&A] [answered]
### Question

How do I deal with a deadlock when trying to deploy a migration to PG? The migration is adding a new column to a few tables that are being constantly used, do I need to split it into 4 smaller calls?

```ts
Database error:
ERROR: deadlock detected
DETAIL: Process 2956293 waits for AccessExclusiveLock on relation 32312 of database 16413; blocked by process 2934826.
Process 2934826 waits for AccessShareLock on relation 26652 of database 16413; blocked by process 2956293.
HINT: See server log for query details.
```
This is the migration
```ts
-- AlterTable
ALTER TABLE "companies" ADD COLUMN     "allowsDynamicOverbooking" BOOLEAN NOT NULL DEFAULT true;

-- AlterTable
ALTER TABLE "regions" ADD COLUMN     "overbookMultiplier" DOUBLE PRECISION NOT NULL DEFAULT 1.00;

-- AlterTable
ALTER TABLE "shift_requests" ADD COLUMN     "enableDynamicOverbooking" BOOLEAN;

-- AlterTable
ALTER TABLE "shifts" ADD COLUMN     "enableDynamicOverbooking" BOOLEAN NOT NULL DEFAULT true;
```

### How to reproduce (optional)

When applying the migration to production, this error happened, it applied fine in our staging environment.

**Comment by janpio** [comment, marked as answer]
Yes, you probably want to apply the usual mechanisms to avoid deadlocks during migrations here. Maybe it is faster to add the column as a NULLable one, and then later modify it when it already exists? I am no expert here (and Prisma unfortunately indeed does not help much with that yet).

**Reply by MorenoMdz** [reply]
Thanks, for context we solved it by individually applying each one of the alter table ops then setting the migration as complete with the Prisma CLI.

**Reply by electrovir** [reply]
I ran into the same problem and the same solution (manually and gradually applying the migration, then marking it as applied) worked for me.

**Comment by HendrikJan** [comment]
Two other things we tried (one of the two seems to have fixed it for us):

We changed the order of the "ALTER TABLE" statements.
I did notice that the two instances of the error that we had, occurred in a table with a relation with the previous statement.
We changed the order of statements such that two subsequent statements never operated on tables with a relation between them.

We also added "sleep 10" (bash command) after de migration and before starting our test (as the error occurred in a CI test).
It might be that "npx prisma migrate reset -f" was still running when the test started.
